hcl/hcl.go

```diff
@@ -221,12 +283,20 @@ func (p *TerramateParser) Parse() (Config, error) {
 // MinimalParse does the syntax parsing and merging of configurations but do not
 // validate if it's valid terramate configuration.
 func (p *TerramateParser) MinimalParse() error {
+	if p.hasInvalidState() {
+		return errors.E("files already parsed")
```

Comment:
hmm shouldn't we just use a field called "parsed" for this then ? Im kinda confused with the "executed" + hasInvalidState + setInvalidState stuff :thinking:

Reply:
hmm I think it is more clear, like you checked if it is already parsed and then returned an error saying "file already parsed". What was odd is that you where checking for an invalid state but then the error message says "file already parsed"...there seems to be a mismatch between the error message and the check itself :thinking: